### PR TITLE
✅ 360 component visual diff test

### PR DESCRIPTION
**Pull request by processprocess**
Visual diff test for rendering a 360 image.

cc @ampproject/wg-stories

### CODE REVIEW

**Comment by gmajoulet** [comment]
image

Do you know why it's not loading on Percy?

**Comment by processprocess** [comment]
> Do you know why it's not loading on Percy?

I think it's not waiting until the image loads. I'll look into the `loading_complete_selectors`

File: extensions/amp-story-360/0.1/amp-story-360.js [hunk 1]
```diff
@@ -725,13 +726,21 @@ export class AmpStory360 extends AMP.BaseElement {
...
+          this.addClassForVisualTests_();
         },
         () => {
           user().error(TAG, 'Failed to load the amp-video.');
         }
       );
   }
 
+  /** @private */
+  addClassForVisualTests_() {
```
Comment (gmajoulet) [inline extensions/amp-story-360/0.1/amp-story-360.js:738]: We usually never ship code that is only meant to help testing, but we can always rename this method and make it reusable in the future for other use cases.

Maybe we can add a `*-loaded` class in a `markAsLoaded_()` method to replicate `amp-story.js` behavior?

**Review by gmajoulet — COMMENTED** [review]
Can we edit the assets to not bloat the AMPHTML repository? I'm worried about every single AMPHTML engineer having to download these when they clone the repository.

The `png` can be heavily optimized, even if it ends up becoming pretty low quality. As long as it renders we're good.
For the video, can we edit it to just keep the first half second or something?

**Comment by processprocess** [comment]
@danielrozenberg As we've discussed, WebGL might not be rendering in Percy.
A screenshot from Puppeteer renders correctly:
Screen Shot 2020-12-01 at 3 15 16 PM

**Review by gmajoulet — APPROVED** [review]

**Comment by processprocess** [comment]
cc @danielrozenberg & @erwinmombay for owners approval.

File: build-system/tasks/visual-diff/snippets/freeze-canvas-image.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+const canvases = document.querySelectorAll('canvas');
+canvases.forEach((canvas) => {
+  const parentNode = canvas.parentElement;
+  const img = document.createElement('img');
+  img.style.width = '100%';
+  img.style.height = '100%';
+  img.setAttribute('src', canvas.toDataURL());
+  parentNode.appendChild(img);
+  parentNode.removeChild(canvas);
```
Comment (danielrozenberg) [inline build-system/tasks/visual-diff/snippets/freeze-canvas-image.js:13]: nit: `parentNode.replaceChild(img, canvas)` (or maybe even `canvas.replaceWith(img)` and completely skip the parentNode, if `canvas` is a ChildNode instance, dunno if that's true though)

**Review by danielrozenberg — APPROVED** [review]

**Review by erwinmombay — APPROVED** [review]

**Pull request merged**